File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_235632/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

function test(description: string, price: number, quantity: number, expected: number): void {
    const got = calculateDiscount(price, quantity);
    const tolerance = 1e-9;
    if (Math.abs(got - expected) < tolerance) {
        console.log(`PASS: ${description}`);
    } else {
        console.log(`FAIL: ${description} (expected: ${expected}, got: ${got})`);
    }
}

// Boundary and key test cases
test("quantity 9 (below 10, no discount)", 10, 9, 10 * 9);
test("quantity 10 (boundary, 10% discount)", 10, 10, 10 * 10 * 0.9);
test("quantity 49 (below 50, 10% discount)", 10, 49, 10 * 49 * 0.9);
test("quantity 50 (boundary, 25% discount)", 10, 50, 10 * 50 * 0.75);
test("quantity 1 (minimum, no discount)", 5, 1, 5 * 1);
test("quantity 100 (above 50, 25% discount)", 3, 100, 3 * 100 * 0.75);

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_235632/go/005_unit_test_writer/solution.go
package main

import "fmt"

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	type testCase struct {
		description string
		price       float64
		quantity    int
		expected    float64
	}

	tests := []testCase{
		{
			description: "quantity 9 (below 10, no discount)",
			price:       10,
			quantity:    9,
			expected:    90,
		},
		{
			description: "quantity 10 (boundary, 10% discount)",
			price:       10,
			quantity:    10,
			expected:    90,
		},
		{
			description: "quantity 49 (below 50, 10% discount)",
			price:       10,
			quantity:    49,
			expected:    441,
		},
		{
			description: "quantity 50 (boundary, 25% discount)",
			price:       10,
			quantity:    50,
			expected:    375,
		},
	}

	for _, tc := range tests {
		got := calculateDiscount(tc.price, tc.quantity)
		if got == tc.expected {
			fmt.Printf("PASS: %s\n", tc.description)
		} else {
			fmt.Printf("FAIL: %s (expected: %.2f, got: %.2f)\n", tc.description, tc.expected, got)
		}
	}
}